components/mitsubishi_itp/mitsubishi_itp-packetprocessing.cpp

```diff
@@ -344,15 +344,9 @@ void MitsubishiUART::handle_thermostat_state_download_request(const GetRequestPa
   response.set_cool_setpoint(this->mhk_state_.cool_setpoint_);
 
 #ifdef USE_TIME
-  if (this->time_source_ != nullptr) {
+  if (this->time_sync_) {
     response.set_timestamp(this->time_source_->now().timestamp);
-  } else {
-    ESP_LOGW(TAG, "No time source specified. Cannot provide accurate time!");
-    response.set_timestamp(1704067200);  // 2024-01-01 00:00:00Z
   }
```

Comment:
`time_sync_` does seem like a better check, but I still think it would be helpful to log a warning if the time hasn't synced. Do you know what the behavior of the MHK2 is if the response is sent with no timestamp set at all (versus sending a default)?

Reply:
I ended up going a different direction: if we don't have a synced time source, then just send the StateDownloadRequest to the heatpump and let it respond (or not) as it chooses. Effectively then the MUART device disappears from the path for handling StateDownloadRequest packets until it is able to respond fully.

Reply:
Heat pumps don't respond to StateDownloadRequest (or at least we haven't seen any that do), it's a special Kumo packet. I suspect that if we fail to respond to any of the mysterious Kumo packets, the MHK2 might decide there isn't a Kumo and stop reporting the extra Kumo data (but I have no idea if this is the case without actually testing it). I don't know if the MHK2 will occasionally check to see if a Kumo has appeared since there's no reason one should suddenly exist without a power cycle of the thermostat receiver.

I can test all this if I get a chance though.

Reply:
Alright... you've convinced me that the fake timestamp is the way to go.